lib/widgets/store.dart

```diff
@@ -195,18 +205,42 @@ class _PerAccountStoreWidgetState extends State<PerAccountStoreWidget> {
   void didChangeDependencies() {
     super.didChangeDependencies();
     final globalStore = GlobalStoreWidget.of(context);
+    final accountExists = globalStore.getAccount(widget.accountId) != null;
+    if (!accountExists) {
+      // logged out
+      _setStore(null);
+      if (widget.routeToRemoveOnLogout != null) {
+        SchedulerBinding.instance.addPostFrameCallback(
+          (_) => Navigator.of(context).removeRoute(widget.routeToRemoveOnLogout!));
+      }
+      return;
+    }
     // If we already have data, get it immediately. This avoids showing one
     // frame of loading indicator each time we have a new PerAccountStoreWidget.
     final store = globalStore.perAccountSync(widget.accountId);
     if (store != null) {
       _setStore(store);
     } else {
       // If we don't already have data, request it.
-      globalStore.perAccount(widget.accountId);
+      () async {
+        try {
+          // If this succeeds, globalStore will notify listeners, and
+          // [didChangeDependencies] will run again, this time in the
+          // `store != null` case above.
+          await globalStore.perAccount(widget.accountId);
+        } on AccountNotFoundException {
+          // The account was logged out while its store was loading.
+          // This widget will be showing [placeholder] perpetually,
+          // but that's OK as long as other code will be removing it from the UI
+          // (usually by using [routeToRemoveOnLogout]).
+        } catch (e) {
+          // TODO(log)
+        }
```

Comment:
```suggestion
        }
```

If we don't catch the exception, it'll become an unhandled exception (because nothing awaits the future returned by this IIFE, and the exception will cause that future to complete with an error). That'll automatically cause it to get logged, once we have any kind of generic logging in place.

Could also say something like
```dart
          rethrow; // TODO(log)
```

but, apart from playing host to the comment, that has exactly the same effect as not having the `catch` block in the first place. Because I think we don't need the TODO, that makes it redundant.